beginner_source/scaled_dot_product_attention_tutorial.py

```diff
@@ -0,0 +1,322 @@
+"""
+Implement High-Performance Transformers with SCALED DOT PRODUCT ATTENTION
+================================================================================
+
+"""
+
+
+######################################################################
+# Summary
+# ~~~~~~~~
+#
+# In this tutorial, we want to highlight a new ``torch.nn.functional`` function
+# that can be helpful for implementing transformer architectures. The
+# function is named ``torch.nn.functional.scaled_dot_product_attention``.
+# For detailed description of the function, see the `PyTorch# documentation <https://pytorch.org/docs/master/generated/torch.nn.functional.scaled_dot_product_attention.html#torch.nn.functional.scaled_dot_product_attention>`__.
+# This function has already been incorporated into ``torch.nn.MultiheadAttention`` and ``torch.nn.TransformerEncoderLayer``.
+#
+# Overview
+# ~~~~~~~~~
+# At a high level, this PyTorch function calculates the
+# scaled dot product attention (SDPA) between query, key, and value according to
+# the definition found in the paper `Attention is all you
+# need <https://arxiv.org/abs/1706.03762>`__. While this function can be
+# written in PyTorch using existing functions, for GPU tensors this
+# function will implicitly dispatch to an optimized implementation. The
+# function is also highly modular and can be used to implement other
```

Comment:
So if you're making ease extensibility claims it's better to prove it by either showing the example or linking to an example that users could plausibly modify to get linformer

Reply:
This is a good point. Linformer uses SDPA as a component in its architecture.  The hop was by building the CasualAttention module that could be modified to create a linformer. That being said I can remove this line